pages/topics/edit.tsx

```diff
@@ -44,13 +46,15 @@ function Edit({ topicId, back, onDelete }: EditProps) {
   }
   async function handleSubtitleSubmit(videoTrack: VideoTrackProps) {
     if (!topic) return;
-    await uploadVideoTrack(topic.resource.id, videoTrack);
+    const uploaded = await uploadVideoTrack(topic.resource.id, videoTrack);
     await revalidateTopic(topic.id);
+    addVideoTrack(uploaded);
   }
-  async function handleSubtitleDelete({ id }: Pick<VideoTrackSchema, "id">) {
+  async function handleSubtitleDelete(videoTrack: VideoTrackSchema) {
     if (!topic) return;
-    await destroyVideoTrack(topic.resource.id, id);
+    await destroyVideoTrack(topic.resource.id, videoTrack.id);
     await revalidateTopic(topic.id);
+    deleteVideoTrack(videoTrack);
```

Comment:
> This condition will always return 'false' since the types 'number' and 'Pick<VideoTrackSchema, "id">' have no overlap. ts(2367)

`Pick<VideoTrackSchema, "id">` だとうまく型推論が働かないようです

Reply:
deleteVideoTrack の型注釈を VideoTrackSchema から Pick<VideoTrackSchema, "id"> に変更してもでしょうか?

Reply:
手元で試した変更は、deleteVideoTrackの引数をPick<VideoTrackSchema, "id">に変更したものになります

```ts
const deleteVideoTrackAtom = atom<null, Pick<VideoTrackSchema, "id">>(
  null,
  (get, set, id) => {
    const videoTracksProps = get(videoTracksPropsAtom);
    const videoTracks = get(videoTracksAtom);
    const indexOrId = videoTracks.findIndex(
      (videoTrack) => videoTrack.id == id
    );
(snip)
```

Reply:
```diff
-const deleteVideoTrackAtom = atom<null, Pick<VideoTrackSchema, "id">>(
+const deleteVideoTrackAtom = atom<null, VideoTrackSchema["id"]>(
```

でしょうかね…?

Reply:
そういえば Pick を使うのと T[key] の使い分けを自分がよく分かっていなさそうですね…それでいけそうな気がします

Reply:
`{ id }` が `Pick<VideoTrackSchema, "id>` で `id` が `VideoTrackSchema["id"]` なので普通に違うものですね…読めば分かることが分かっていなかった…